pkg/agent/proxy/proxier.go

```diff
@@ -366,15 +408,27 @@ func smallSliceDifference(s1, s2 []string) []string {
 	return diff
 }
 
-func (p *proxier) installNodePortService(externalGroupID, clusterGroupID binding.GroupIDType, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16) error {
+func (p *proxier) installNodePortService(localGroupID, clusterGroupID binding.GroupIDType, svcPort uint16, protocol binding.Protocol, trafficPolicyLocal bool, affinityTimeout uint16) error {
 	if svcPort == 0 {
 		return nil
 	}
 	svcIP := agentconfig.VirtualNodePortDNATIPv4
 	if p.isIPv6 {
 		svcIP = agentconfig.VirtualNodePortDNATIPv6
 	}
-	if err := p.ofClient.InstallServiceFlows(externalGroupID, clusterGroupID, svcIP, svcPort, protocol, affinityTimeout, true, false); err != nil {
+	if err := p.ofClient.InstallServiceFlows(&agenttypes.ServiceConfig{
+		ServiceIP:          svcIP,
+		ServicePort:        svcPort,
+		Protocol:           protocol,
+		TrafficPolicyLocal: trafficPolicyLocal,
+		LocalGroupID:       localGroupID,
+		ClusterGroupID:     clusterGroupID,
+		AffinityTimeout:    affinityTimeout,
+		IsExternal:         true,
+		IsNodePort:         true,
+		IsNested:           false, // Unsupported for NodePort
+		IsDSR:              false, // Unsupported for NodePort
```

Comment:
I'm curious as to why we are not also supporting DSR for NodePort Service access?